datasync/syncbase/watcher.go

```diff
@@ -100,39 +101,49 @@ func (adapter *Registry) PropagateChanges(txData map[string]datasync.ChangeValue
 	var events []func(done chan error)
 
 	for _, sub := range adapter.subscriptions {
+		var changes []datasync.ProtoWatchResp
```

Comment:
On more thing: I wonder if it would be possible to propagate upwards a list of key-error pairs instead of just one error? For example in resync multiple values may fail - currently we return only one of the errors.